### PR TITLE
WIP: Add Zora

### PR DESCRIPTION
**Pull request by vicsn**
This PR is a proof of concept Zora integration. It partially implements: 
- https://github.com/721labs/partial-common-ownership/issues/47
- https://github.com/721labs/partial-common-ownership/issues/42

There is a lot still left to do:
- [ ] if the NFT is sold on Zora (in ZoraSpeak, the Ask is filled) then `takeoverLease(...)` should be called somehow with a new valuation. But Zora's `fillAsk(...)` function doesn't offer the right interface. So for any exchange we integrate, a "harberger-specific" storefront has to be built. I'll have a look at the Zora ecosystem to see what that would look like - maybe existing websites are open to support the interface.
- [ ] For every function of the base partialcommonownership contract, we need to review how it might impact the listing. For example, tokens could be delisted when foreclosed. One way to approach this is to add a modifier to functions which includes logic like: `if (listed) { do_something_with_listing() }`
- [ ] Cleanup modifiers, parameter checks, event logs, comments, proper interface inheritance, etc
- [ ] Use Zora's [production addresses](https://docs.zora.co/docs/smart-contracts/modules/Asks/zora-v3-asks-latest)
- [ ] Think about whether we will need "friendly transfers" towards exchange contracts: https://github.com/721labs/partial-common-ownership/issues/35 . We may be able to ignore this for now.

For now I have the following question: how should we test the integration? I wanted to deploy all Zora contracts to see how our contracts interact with them. However, not all Zora contracts are available (or at least, documented...) in the `@zoralabs`' packages. Therefore I added the entire Zora/v3 repo as a submodule now. It feels a bit redundant, but on the other hand it ensures we learn how the exchange contracts work.

### CODE REVIEW

File: contracts/Wrapper.sol [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // contracts/Wrapper.sol
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.7;
+pragma solidity 0.8.10;
```
[the remark this answers is not in the document]
Comment (will-holley) [reply contracts/Wrapper.sol:3]: I went ahead and upgraded to the latest Solidity compiler (`0.8.12`).  Please pull `main` in!

**Comment by will-holley** [comment]
I'm looking forward to reviewing this.

As to your questions, I feel like @gravenp and @codynhat also have constructive ideas here.

**Comment by gravenp** [comment]
Haven't had a chance to review yet, but just came across this Harberger NFT that just launched and is using Zora: https://harberger.tax/

I'm trying to connect with the builders now.

**Comment by will-holley** [comment]
@gravenp  I can't tell if it's broken or if their flavor isn't a perpetual auction.

**Comment by gravenp** [comment]
I was just on a Twitter space with them and they're having troubles with pushing it live, so "broken." It appears that they're going for a periodic auction style which the owner manually(?) relists the item. Definitely a much different approach than the applications that I've seen, but it was likely forced by their desired to use Zora "now."

**Comment by iainnash** [comment]
Hey! What packages are missing in the node packages?

If there’s a custom type of market module you want to propose that’s also an option. 

We’re working on front end modules for purchasing and selling NFTs within a modal.

**Pull request closed without merging**